libs/shared/otel/src/lib/exporters/exporters.ts

```diff
@@ -8,18 +8,31 @@ import {
   BatchSpanProcessor,
   SimpleSpanProcessor,
   SpanExporter,
+  SDKRegistrationConfig,
 } from '@opentelemetry/sdk-trace-node';
 
+/**
+ * Adds an exporter to the provider. If options.batchProcessor is true, a BatchSpanProcessor
+ * is used, otherwise a SimpleSpanProcessor is used.
+ * @param opts - The tracing options
+ * @param provider - The tracer provider
+ * @param exporter - The span exporter to add
+ * @returns The span processor that was added
+ */
 export function addExporter(
   opts: TracingOpts,
   provider: NodeTracerProvider,
   exporter: SpanExporter
 ) {
-  const processor = opts.batchProcessor
+  const spanProcessor = opts.batchProcessor
     ? new BatchSpanProcessor(exporter)
     : new SimpleSpanProcessor(exporter);
-  provider.addSpanProcessor(processor);
 
-  provider.addSpanProcessor(processor);
```

Comment:
Welp, it was definitely a mistake that there were two of these 'addProcessor' calls here. Thanks for cleaning that up!